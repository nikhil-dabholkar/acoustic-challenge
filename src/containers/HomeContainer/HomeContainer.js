import React from 'react';
import { lazy, Suspense } from 'react';
import { LoadingComponent } from '../../components/index';

const HomeComponent = lazy(() => import("../../components/home/HomeComponent/HomeComponent"));
// import {HomeComponent} from '../../components/index';

function HomeContainer() {
    return (<React.Fragment>
        <div data-testid="homeContainer">
            <Suspense fallback={<LoadingComponent />}>
                <HomeComponent></HomeComponent>
            </Suspense>
        </div>
    </React.Fragment>
    );
}

export default HomeContainer;
