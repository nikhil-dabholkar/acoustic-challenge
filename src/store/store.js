import rootReducer from '../reducer/index';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';

export const store = createStore(rootReducer, applyMiddleware(thunk))