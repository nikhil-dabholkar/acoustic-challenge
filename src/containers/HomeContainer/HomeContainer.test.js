import React from 'react'
import HomeContainer from './HomeContainer'
import { IntlProvider } from 'react-intl'
import { store } from '../../store/store'
import { Provider } from 'react-redux'
import { render, cleanup, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import enUS from '../../i18n/en_US.json'

afterEach(cleanup)

function renderWithRouter (
  ui,
  { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {}
) {
  return {
    ...render(<Router history={history}>{ui}</Router>),
    history
  }
}

function rendererWithRouter (
  ui,
  { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {}
) {
  return {
    ...renderer.create(<Router history={history}>{ui}</Router>),
    history
  }
}

describe('HomeContainer', () => {
  test('Check if HomeContainer Renders on /home', async () => {
    const route = '/home'
    renderWithRouter(<Provider store={store}><IntlProvider locale="en" messages={enUS}><HomeContainer></HomeContainer></IntlProvider></Provider>, { route })
    const lazyElement = await screen.findByTestId('homeContainer')
    expect(lazyElement).toBeInTheDocument()
  })

  test('Check if HomeContainer Renders on /', async () => {
    const route = '/'
    renderWithRouter(<Provider store={store}><IntlProvider locale="en" messages={enUS}><HomeContainer></HomeContainer></IntlProvider></Provider>, { route })
    const lazyElement = await screen.findByTestId('homeContainer')
    expect(lazyElement).toBeInTheDocument()
  })

  test('Check if matches HomeContainer Snapshot', async () => {
    const route = '/home'
    const tree = rendererWithRouter(<Provider store={store}><IntlProvider locale="en" messages={enUS}><HomeContainer></HomeContainer></IntlProvider></Provider>, { route }).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
