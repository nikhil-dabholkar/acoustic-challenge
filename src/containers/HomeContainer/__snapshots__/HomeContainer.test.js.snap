// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContainer Check if matches HomeContainer Snapshot 1`] = `
<div
  data-testid="homeContainer"
>
  <div
    data-testid="homeComponent"
  >
    <img
      src="https://i2.wp.com/spyderslab.com/wp-content/uploads/2018/06/digital_marketing_spyderslab.jpg?resize=1170%2C480&ssl=1"
    />
  </div>
</div>
`;
